tests/phpunit/CRM/Utils/Mail/EmailProcessorTest.php

```diff
@@ -69,6 +69,21 @@ public function testProcessingMultipartRelatedEmail() {
     $this->checkMailingBounces(1);
   }
 
+  /**
+   * Tests that a nested multipart email does not cause pain & misery & fatal errors.
+   *
+   * Sample anonymized from an email that broke bounce processing at Wikimedia
+   */
+  public function testProcessingNestedMultipartEmail() {
+    $this->setUpMailing();
+    $mail = 'test_nested_message.eml';
+
+    copy(__DIR__ . '/data/bounces/' . $mail, __DIR__ . '/data/mail/' .   $mail);
+    $this->callAPISuccess('job', 'fetch_bounces', array());
+    $this->assertFalse(file_exists(__DIR__ . '/data/mail/' . $mail));
+    $this->checkMailingBounces(1);
```

Comment:
@ejegg Just wondering if its worth while checking for a specific type and or message stored in the db associated with the bounce also?